File: Arcade/TheCore/Java/CountSumOfTwoRepresentations2.java
final class CountSumOfTwoRepresentations2 {
	
	int countSumOfTwoRepresentations2(int n, int l, int r) {
	    if (2 * r < n || 2 * l > n) return 0;
	    double min = Math.max(l, n - r);
	    double max = Math.min(r, n - l);
	    return (int)(Math.floor((max + min) / 2) - min) + 1;
	}
}

File: Arcade/TheCore/Kotlin/countSumOfTwoRepresentations2.kt
fun countSumOfTwoRepresentations2(n: Int, l: Int, r: Int): Int {
    if (2 * r < n || 2 * l > n) return 0;
    val min: Double = Math.max(l.toDouble(), n.toDouble() - r.toDouble());
    val max: Double = Math.min(r.toDouble(), n.toDouble() - l.toDouble());
    return ((Math.floor((max + min) / 2.0) - min) + 1.0).toInt();
}